src/analyze/flavors/lit-element/parse-lit-property-configuration.ts

```diff
@@ -61,124 +61,124 @@ export function getLitElementPropertyDecoratorConfig(node: Node, context: Analyz
-		const kind = ts.isIdentifier(property.name) ? property.name.text : undefined;
-
-		return parseLitPropertyOption({ kind, initializer, config }, context);
-	}, existingConfig);
-}
-
-export function parseLitPropertyOption(
-	{ kind, initializer, config }: { kind: string | undefined; initializer: Expression; config: LitElementPropertyConfig },
-	context: AnalyzerVisitContext
-): LitElementPropertyConfig {
-	const { ts, checker } = context;
-
-	// noinspection DuplicateCaseLabelJS
-	switch (kind) {
-		case "converter": {
-			return { ...config, hasConverter: true };
+	if (typeof object === "object" && object !== null && !Array.isArray(object)) {
+		if (hasProperty(object, "converter") && object.converter !== undefined) {
+			result.hasConverter = true;
 		}
 
-		case "reflect": {
-			return { ...config, reflect: resolveNodeValue(initializer, context)?.value === true };
+		if (hasProperty(object, "reflect") && object.reflect !== undefined) {
+			result.reflect = object.reflect === true;
 		}
 
-		case "state": {
-			return { ...config, state: resolveNodeValue(initializer, context)?.value === true };
+		if (hasProperty(object, "state") && object.state !== undefined) {
+			result.state = object.state === true;
 		}
 
-		case "attribute": {
-			let attribute: LitElementPropertyConfig["attribute"] | undefined;
-
-			if (initializer.kind === ts.SyntaxKind.TrueKeyword) {
-				attribute = true;
-			} else if (initializer.kind === ts.SyntaxKind.FalseKeyword) {
-				attribute = false;
-			} else if (ts.isStringLiteral(initializer)) {
-				attribute = initializer.text;
-			}
-
-			return {
-				...config,
-				attribute,
-				node: {
-					...(config.node || {}),
-					attribute: initializer
-				}
-			};
+		if (hasProperty(object, "value")) {
+			result.default = object.value;
 		}
 
-		case "type": {
-			let type: LitElementPropertyConfig["type"] | undefined;
-			const value = ts.isIdentifier(initializer) ? initializer.text : undefined;
-
-			switch (value) {
-				case "String":
-				case "StringConstructor":
-					type = { kind: "STRING" };
-					break;
-				case "Number":
-				case "NumberConstructor":
-					type = { kind: "NUMBER" };
-					break;
-				case "Boolean":
-				case "BooleanConstructor":
-					type = { kind: "BOOLEAN" };
-					break;
-				case "Array":
-				case "ArrayConstructor":
-					type = { kind: "ARRAY", type: { kind: "ANY" } };
-					break;
-				case "Object":
-				case "ObjectConstructor":
-					type = { kind: "OBJECT", members: [] };
-					break;
-				default:
-					// This is an unknown type, so set the name as a string
-					type = initializer.getText();
-					break;
-			}
+		if (hasProperty(object, "attribute") && (typeof object.attribute === "boolean" || typeof object.attribute === "string")) {
+			result.attribute = object.attribute;
 
-			return {
-				...config,
-				type,
-				node: {
-					...(config.node || {}),
-					type: initializer
+			if (ts.isObjectLiteralExpression(node)) {
```

Comment:
its trying to handle `attr: "name"`, for example:

```ts
@property({attribute: 'hello-there'})
public helloThere: string;
```

would be `<x-foo hello-there="123">` in dom.

i am wondering though if i can constrain this function as a whole to accept object literal expressions, though. not sure there's a need to ever accept anything else

Reply:
ah sorry just to correct myself a bit. its so we can get hold of the initialiser! i was talking nonsense before.

basically, `resolveNodeValue` resolves two things:

* `value` - the constant value we've determined it will be at runtime
* `node` - the actual AST node we ended up resolving (following variable references)

that means `value.attribute` will be the literal string `hello-there` in my example above.

but to expose the initialiser (the ast node of that), we need to lookup the key in the `node` and get its value as a node (rather than resolving it into a value).

Reply:
Oh, I see, the initializer is the TS node for the value of the "attribute" key in the property config object? Weird name for it, but ok!

Reply:
yeah exactly, could do with a much better name at some point i think :D